tutorials/scripting/gdextension/what_is_gdextension.rst

```diff
@@ -0,0 +1,103 @@
+  languages.
+- You can use the same compiled GDExtension library in the editor and exported
+  project. With C++ modules, you have to recompile all the export templates you
+  plan to use if you require its functionality at run-time.
+- GDExtension only requires you to compile your library, not the whole engine.
+  That's unlike C++ modules, which are statically compiled into the engine.
+  Every time you change a module, you need to recompile the engine. Even with
+  incremental builds, this process is slower than using GDExtension.
+
+Advantages of C++ modules
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We recommend :ref:`C++ modules <doc_custom_modules_in_c++>` in cases where
+GDExtension isn't enough:
+
+- C++ modules provide deeper integration into the engine. GDExtension's access is not as deep as
+  static modules
+- You can use C++ modules to provide additional features in a project without
+  carrying native library files around. This extends to exported projects.
+(- C++ modules can be faster than GDNative, especially when the code requires a
+  lot of communication through the scripting API.) -> TODO: check if this still holds up
+
+Supported languages
+-------------------
+
+The Godot developers officially support the following language bindings for
+GDExtension:
+
+- C++ :ref:`(tutorial) <doc_gdnative_cpp_example>`
+- C :ref:`(tutorial) <doc_gdnative_c_example>`
+
+.. note::
+
+    There are no plans to support additional languages with GDExtension officially.
+    That said, the community offers several bindings for other languages (see
+    below).
+
+.. _doc_what_is_gdnative_third_party_bindings:
+
+The bindings below are developed and maintained by the community:
+
+.. Binding developers: Feel free to open a pull request to add your binding if it's well-developed enough to be used in a project.
+.. Please keep languages sorted in alphabetical order.
+
+- `D <https://github.com/godot-d/godot-d>`__
+- `Kotlin <https://github.com/utopia-rise/godot-kotlin-jvm>`__
+- `Nim <https://github.com/pragmagic/godot-nim>`__
+- `Python <https://github.com/touilleMan/godot-python>`__
+- `Rust <https://github.com/godot-rust/godot-rust>`__
+
+.. note::
+
+    Not all bindings mentioned here may be production-ready. Make sure to
+    research options thoroughly before starting a project with one of those.
+    Also, double-check whether the binding is compatible with the Godot version
+    you're using.
+
+Version compatibility
+---------------------
+
+(:ref:`Unlike Godot itself <doc_release_policy>`, GDExtension has stricter version
+compatibility requirements as it relies on low-level *ptrcalls* to function.) -> TODO: check up on this
+
+GDExtension add-ons compiled for a given Godot version are only guaranteed to work
+with the same minor release series. For example, a GDExtension add-on compiled for
+Godot 4.0 will only work with Godot 4.0, 4.0.1, 4.0.2… but not Godot 3.5 or 3.6.
```

Comment:
Nitpick: I'd rephrase to make it clearer that 3.5 and 3.6 don't have GDExtension at all (unless I missed something, GDNative was NOT backported to 3.x branch)

Reply:
Good point👍🏻 I made it clearer that GDExtension isn't compatible with GDNative